modules/task_manager/cycle_routing_manager.cc

```diff
@@ -45,18 +45,26 @@ common::Status CycleRoutingManager::Init(
 
 bool CycleRoutingManager::CheckIfReachDestination(
         const localization::Pose &pose) {
-    if (CheckPointDistanceInThreshold(
-          last_point_,
+    ADEBUG << "Check if reach destination: localization_pose: "
+    << pose.position().x() << " " << pose.position().y()
+    << ", begin point " << begin_point_.x() << " " << begin_point_.y()
+    << ", end point " << end_point_.x() << " " << end_point_.y()
+    << ", threshold " << FLAGS_threshold_for_destination_check
+    << ", allowed_to_send_routing_request " << is_allowed_to_route_;
+
+    if (!CheckPointDistanceInThreshold(
           begin_point_,
-          FLAGS_threshold_for_destination_check)) {
-        last_point_ = pose.position();
+          pose.position(),
+          FLAGS_threshold_for_destination_check * 2)) {
```

Comment:
Why double it here?

Reply:
In case of localization pose drift, routing is only allowed when the pose is 2*Threshold away from the beginning position. I will change 2 to a FLAG.